.github/workflows/release.yml

```diff
@@ -0,0 +1,145 @@
+name: release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  draft-release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+      - name: Install Carvel Tools
+        uses: vmware-tanzu/carvel-setup-action@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          only: ytt, kapp, kbld, imgpkg
+          ytt: v0.40.1
+          kapp: v0.47.0
+          kbld: v0.33.0
+          imgpkg: v0.29.0
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Run release script
+        run: |
+          set -e -x
+          minikube start --driver=docker --wait=all
+          docker buildx create minikube --use --driver=kubernetes --bootstrap
+          ./hack/build-release.sh
+      - name: Create release draft and upload release yaml
+        uses: softprops/action-gh-release@17cd0d34deddf848fc0e7d9be5202c148c270a0a
+        with:
+          name: ${{ github.ref_name }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          body_path: ./tmp/checksums.txt
+          files: |
+            ./tmp/release.yml
+            ./tmp/package.yml
+            ./tmp/metadata.yml
```

Comment:
might be worth calling this `package-metadata.yml` (metadata.yml looks a bit out of context without the prefix)